bin/sessionmirror.py

```diff
@@ -0,0 +1,749 @@
+        if op.exists(loc_f) is False:
+            src_f.get(loc_f)
+
+        # Get File Attributes
+        f_in_attrs = src_f.attributes()
+        f_content = f_in_attrs.get('file_content')
+        f_format = f_in_attrs.get('file_format')
+        f_tags = f_in_attrs.get('file_tags')
+
+        # Upload File
+        if f_format and f_content and not f_tags:         # format & content
+            dest_r.file(f_label).put(loc_f, f_format, f_content)
+        elif f_format and not f_content and not f_tags:   # format only
+            dest_r.file(f_label).put(loc_f, f_format)
+        elif f_format and f_content and f_tags:   # format, content, & tags
+            dest_r.file(f_label).put(loc_f, f_format, f_content)
+        else:                                             # none
+            dest_r.file(f_label).put(loc_f)
+
+        # Delete local copy
+        os.remove(loc_f)
+    except Exception:
+        print("ERROR:failed to copy file:%s, error=%s"
+              % (f_label, sys.exc_info()[0]))
+
+
+def copy_res_zip(src_r, dest_r, cache_d):
+    '''
+    Copy a resource from XNAT source to XNAT destination using local cache
+    in between
+    '''
+    try:
+        # Download zip of resource
+        print('INFO:Downloading resource as zip...')
+        cache_z = src_r.get(cache_d, extract=False)
+
+        # Upload zip of resource
+        print('INFO:Uploading resource as zip...')
+        dest_r.put_zip(cache_z, extract=True)
+
+        # Delete cached zip
+        os.remove(cache_z)
+
+    except IndexError:
+        print('ERROR:failed to copy:%s:%s' % (cache_z, sys.exc_info()[0]))
+        raise
+
+
+def is_empty_resource(_res):
+    '''Check if resource contains any files'''
+    f_count = 0
+    for f_in in _res.files().fetchall('obj'):
+        f_count += 1
+        break
+
+    return f_count == 0
+
+
+def copy_project(src_proj, dst_proj, proj_cache_dir):
+    '''Copy XNAT project from source to destination'''
+
+    if not dst_proj.exists():
+        try:
+            dst_proj.create()
+        except Exception as e:
+            msg = 'ERROR: can not create project on destination '\
+                'xnat. Check your user rights. %s' % e
+            raise Exception(msg)
+        copy_attributes(src_proj, dst_proj)
+    print('INFO:loading and sorting subject list...')
+
+    proj_label = src_proj.label()
+    subj_list = src_xnat.get_subjects(proj_label)
+    subj_list = [x for x in subj_list if x['label'] not in SUBJECTS_MIRRORED]
+    subj_i = 0
+    for subj in subj_list:
+        subj_i += 1
+        subject_label = subj['label']
+
+        print("INFO:Processing subject %s (%d)..." % (subject_label, subj_i))
+        src_subj = src_proj.subject(subject_label)
+        dst_subj = dst_proj.subject(subject_label)
+        subj_cache_dir = op.join(proj_cache_dir, subject_label)
+        copy_subject(src_subj, dst_subj, subj_cache_dir)
+
+
+def copy_subject(src_subj, dst_subj, subj_cache_dir):
```

Comment:
Note that `copy_subject` function does not copy subject-level resources

Reply:
Same comment as for `copy_project`.